OP5 (source-repos): Auto Fetching examples added with variables in CI (#541)
Rookout/docs: docs/source-repos.md

<iframe width="600" height="300" src="https://www.youtube.com/embed/d4LwqNFeR7s" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

#### Automatic Fetching CI Examples

> As an example of how to incorporate autofetching into a Docker application pipeline, you should add env variables with arguments to your `Dockerfile` as illustrated below, and pass those arguments using CI variables with docker build command. This way, it will pass actual information about the origin commit to rookout system. See table with predefined variables in different CI systems and example pipeline steps.

Reviewer: ```suggestion
As an example of how to incorporate auto fetching into a Docker application pipeline, you should add environment variables with arguments to your `Dockerfile` as illustrated below, then pass those arguments using CI variables with the `docker build` command. See table with predefined variables in different CI systems and example pipeline steps.
```

Author: done

| GitHub Actions | github.repositoryUrl <br> GITHUB_SHA <br> github.sha | https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables <br> https://docs.github.com/en/actions/learn-github-actions/contexts |
| Azure DevOps | Build.SourceVersion <br> Build.Repository.Uri | https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml |
| BitBucket Pipelines | BITBUCKET_REPO_FULL_NAME <br> BITBUCKET_COMMIT | https://support.atlassian.com/bitbucket-cloud/docs/variables-and-secrets/ |
| GitLab CI | CI_REPOSITORY_URL <br> CI_COMMIT_SHA | https://docs.gitlab.com/ee/ci/variables/predefined_variables.html |
| Travis CI | $(git config --get remote.origin.url) <br> TRAVIS_COMMIT | https://docs.travis-ci.com/user/environment-variables/ |
| BuildBot | repoUrl <br> revision | http://docs.buildbot.net/latest/manual/configuration/changesources.html |
| AWS CodePipeline | CODEBUILD_SOURCE_REPO_URL <br> CODEBUILD_RESOLVED_SOURCE_VERSION | https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-env-vars.html |
| Shell | \$(git rev-parse HEAD) <br> $(git config --get remote.origin.url) | |

> #### Pipeline Step Examples

Reviewer: I think this line should be removed

Author: Done

To make sure you are collecting data from the source line where you have set the breakpoint, include your source files within your library.

For more information, see [this page](dotnet-setup#packaging-sources).

> #

Reviewer: This line should be removed, is there a purpose for it?

Author: done
